### PR TITLE
Replaced reference to obsolete FlatButton button class in SnackBar

### PR DESCRIPTION
**Pull request by HansMuller**
Replaced the one use of FlatButton with TextButton in SnackBar and removed its dependency on ButtonTheme per #59702.

### CODE REVIEW

**Comment by HansMuller** [comment]
This PR doesn't include a test because the substitutions are covered by the existing tests; there's no change in behavior or API here. Although we could verify that SnackBar no longer uses FlatButton, FlatButton will be deprecated soon and that will prevent the unlikely possibility of backsliding.

File: packages/flutter/lib/src/material/snack_bar.dart [hunk 1]
```diff
@@ -445,10 +452,13 @@ class _SnackBarState extends State<SnackBar> {
             ),
           ),
           if (widget.action != null)
-            ButtonTheme(
-              textTheme: ButtonTextTheme.accent,
-              minWidth: 64.0,
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/snack_bar.dart:450]: Is the minWidth already accounted for in the TextButton.styleFrom()?

**Review by darrenaustin — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/snack_bar.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/lib/src/material/snack_bar.dart:450]: Yes, the default minimum size for TextButton is Size(64, 36)

**Pull request merged**